File: Euclid/Messages/Outgoing/ACTIVEOBJECT_REMOVE.cs
using Euclid.Game;
using Euclid.Network.Streams.Util;
using System.Collections.Generic;

namespace Euclid.Messages.Outgoing
{
    public class ACTIVEOBJECT_REMOVE : IMessageComposer
    {
        private Item item;

        public ACTIVEOBJECT_REMOVE(Item item)
        {
            this.item = item;
        }

        public override void Write()
        {
            Data.Add(new ArgumentEntry(this.item.PaddedId));
            Data.Add(new DelimeterEntry(this.item.Data.Id, string.Empty));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/ACTIVEOBJECT_REMOVE.java
package euclid.messages.outgoing;

import euclid.game.item.Item;
import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;

public class ACTIVEOBJECT_REMOVE extends IMessageComposer {

    private final Item item;

    public ACTIVEOBJECT_REMOVE(Item item) {
        this.item = item;
    }

    @Override
    public void write() {
        getData().add(new ArgumentEntry(item.getPaddedId()));
        getData().add(new DelimeterEntry(item.getData().getId(), ""));
    }
}
